apps/frontend/src/helpers/crypto.ts

```diff
@@ -0,0 +1,67 @@
+import { ERC20_EURE_POLYGON_DECIMALS, ERC20_EURE_POLYGON_V2, multiplyByPowerOfTen } from "@vortexfi/shared";
+import { readContract, signTypedData } from "@wagmi/core";
+import { wagmiConfig } from "../wagmiConfig";
+
+export async function signERC2612Permit(
```

Comment:
This function looks like it's generic but it's not really. It's coupled to the EURE contract on polygon and we are assuming that for the values passed. Let's either rename the function to indicate this or make it more generic by allowing to pass a contract address and instead of passing `valueUnits` we should pass `valueRaw`.